@ecomplus/storefront-template/template/pages/@/sections/product-block.ejs

```diff
@@ -237,6 +238,17 @@ if (_.state.gtin && _.state.gtin[0]) {
   productJsonLd.gtin = _.state.gtin[0]
 }
 %>
-<script type="application/ld+json"><%-
+<script id="product-block-jsonld" type="application/ld+json"><%-
   JSON.stringify(productJsonLd)
 %></script>
+<script type="text/javascript">
+  const urlParams = new URLSearchParams(window.location.search)
+  const variationId = urlParams.get('variation_id')
+  const jsonLd = <%- JSON.stringify(productJsonLd) %>
+  const variations = <%- JSON.stringify(_.state.variations) %>
+  if (variationId && Array.isArray(variations) && variations.length) {
+    const selectedVariation = variations.find(variation => variation._id === variationId)
```

Comment:
Mesma coisa do componente Vue, não é `variations.length` que você precisa checar (desnecessário), se for array e existir um id você tenta selecionar a variação, depois você testa `if (selectedVariation) {` se a variação existe..

Reply:
Aqui você vai dar fatal se vier um valor inválido (ou variação inexistente) como parâmetro de URL

Reply:
E é um input de usuário então é perfeitamente factível..